F401: Ignoring errors fails with capitalisation

Ignoring errors on a line-by-line basis should work with any capitalisation of `#noqa`, in my projects I tend to use `#NoQA`.

This failure was introduced in Ruff 111, see below:

```doscon
I:\Development>type bug.py
import this  # NoQA: F401
import antigravity  # noqa: F401

I:\Development>ruff -V     
ruff 0.0.110

I:\Development>ruff bug.py

A new version of ruff is available: v0.0.110 -> v0.0.116
Run to update: pip3 install --upgrade ruff

I:\Development>pip install "ruff==0.0.111"
...
Successfully installed ruff-0.0.111

I:\Development>ruff -V     
ruff 0.0.111

I:\Development>ruff bug.py 
Found 1 error(s).
bug.py:1:1: F401 `this` imported but unused
1 potentially fixable with the --fix option.

A new version of ruff is available: v0.0.111 -> v0.0.116
Run to update: pip3 install --upgrade ruff

I:\Development>
```

A

--- a/src/noqa.rs
+++ b/src/noqa.rs
@@ -10,7 +10,7 @@ use regex::Regex;
 use crate::checks::{Check, CheckCode};
 
 static NO_QA_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(r"(?P<noqa>\s*# noqa(?::\s?(?P<codes>([A-Z]+[0-9]+(?:[,\s]+)?)+))?)")
+    Regex::new(r"(?P<noqa>\s*(?i:# noqa)(?::\s?(?P<codes>([A-Z]+[0-9]+(?:[,\s]+)?)+))?)")
         .expect("Invalid regex")
 });
 static SPLIT_COMMA_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"[,\s]").expect("Invalid regex"));
@@ -118,7 +118,21 @@ mod tests {
 
     use crate::ast::types::Range;
     use crate::checks::{Check, CheckKind};
-    use crate::noqa::add_noqa_inner;
+    use crate::noqa::{add_noqa_inner, NO_QA_REGEX};
+
+    #[test]
+    fn regex() {
+        assert!(NO_QA_REGEX.is_match("# noqa"));
+        assert!(NO_QA_REGEX.is_match("# NoQA"));
+
+        assert!(NO_QA_REGEX.is_match("# noqa: F401"));
+        assert!(NO_QA_REGEX.is_match("# NoQA: F401"));
+        assert!(NO_QA_REGEX.is_match("# noqa: F401, E501"));
+
+        assert!(NO_QA_REGEX.is_match("# noqa:F401"));
+        assert!(NO_QA_REGEX.is_match("# NoQA:F401"));
+        assert!(NO_QA_REGEX.is_match("# noqa:F401, E501"));
+    }
 
     #[test]
     fn modification() -> Result<()> {

